zaza/openstack/utilities/openstack.py

```diff
@@ -2651,10 +2653,68 @@ def upload_image_to_glance(glance, local_path, image_name, disk_format='qcow2',
     return image
 
 
+def is_ceph_image_backend(model_name=None):
+    """Check if glance is related to ceph, therefore using ceph as backend.
+
+    :returns: True if glance is related to Ceph, otherwise False
+    :rtype: bool
+    """
+    status = juju_utils.get_full_juju_status(model_name=model_name)
+    result = False
+    try:
+        result = 'ceph-mon' in (
+            status['applications']['glance']['relations']['ceph'])
+    except KeyError:
+        pass
+    if result is False:
+        logging.info("Did not detect Ceph related to Glance")
+    return result
+
+
+def ensure_image_format(local_path, disk_format):
+    """Convert the image if its format differs from the desired format.
+
+    :param local_path: The path to the original image file.
+    :type local_path: str
+    :param disk_format: The format of the underlying disk image.
+    :type disk_format: str
+    :returns: The path to the final image file
+    :rtype: str
+    """
+    try:
+        output = subprocess.check_output([
+            "qemu-img", "info", "--output=json", local_path])
+    except subprocess.CalledProcessError:
+        logging.error("Image conversion: Failed to detect image format "
+                      "of file {}".format(local_path))
+        return local_path
+    result = json.loads(output)
+    if disk_format == 'raw' and result['format'] == 'qcow2':
+        logging.info("Image conversion: Detected qcow2 vs desired raw format"
+                     " of file {}".format(local_path))
+        converted_path = pathlib.Path(local_path).resolve().with_suffix('.raw')
+        converted_path_str = str(converted_path)
+        if converted_path.exists():
+            logging.info("Image conversion: Desired converted file already"
+                         " exists: {}".format(converted_path_str))
+            return converted_path_str
+        logging.info("Image conversion: Converting image {} to {}".format(
+            local_path, disk_format))
+        try:
+            output = subprocess.check_output([
+                "qemu-img", "convert", local_path, converted_path_str])
+        except subprocess.CalledProcessError:
+            logging.error("Image conversion: Failed to convert image"
+                          " {} to {}".format(local_path, disk_format))
+            return local_path
+        return converted_path_str
+    return local_path
+
+
 def create_image(glance, image_url, image_name, image_cache_dir=None, tags=[],
                  properties=None, backend=None, disk_format='qcow2',
                  visibility='public', container_format='bare',
-                 force_import=False):
+                 force_import=False, convert_if_ceph=True):
```

Comment:
I don't think this is descriptive enough.  It needs to be something like `convert_image_to_raw_if_ceph_used=True`; i.e. explain the purpose in the name of the variable.

Plus this new parameter needs to be explained in the docstring and why it is being done; please refer to the bug being 'fixed' by this for future readers of the code.

Reply:
done, indeed I had forgotten the docstring. I need to log a bug for this so it can be referenced